pages/patient/index.tsx

```diff
@@ -25,21 +26,55 @@ const PatientHome = () => {
         );
         return;
       }
-
       setAppointments(res);
     };
     getAppointments();
   }, [apiClient, user.sub, role]);
 
+  const filterAppointments = (appointments: Appointment[]): Appointment[] => {
+    const currentDate = Date.now();
+    const filteredAppointments = appointments
+      .filter((appointment) => {
+        if (filter === 'Upcoming') {
+          return appointment.appointment_time > currentDate;
+        } else {
+          return appointment.appointment_time < currentDate;
+        }
+      })
+      .sort((a, b) => a.appointment_time - b.appointment_time);
+    console.log(filteredAppointments);
+    return filteredAppointments;
+  };
+
   return (
     <>
       <Navbar />
+      <div className="container flex w-full my-4 mx-auto items-center justify-end">
+        <label className="mx-2">Filter Appointments</label>
+        <select
+          className="
+                    mr-5
+                    block
+                    w-fit
+                    rounded-md
+                    border-gray-300
+                    shadow-sm
+                    focus:border-purple-300 focus:ring focus:ring-purple-200 focus:ring-opacity-50
+                  "
+          value={filter}
+          onChange={(e) => setFilter(e.target.value)}
+        >
+          <option>Upcoming</option>
```

Comment:
Do you need to set the `value` prop here?

Reply:
You right I don't need to set it since I have default value for `useState`